tests/conftest.py

```diff
@@ -199,10 +199,10 @@ def reset(self):
         self._messages = defaultdict(list)
         self._offsets = defaultdict(dict)
 
-    def write(self, topic, msg, headers=None):
+    def write(self, topic, msg, headers=[]):
```

Comment:
For this and the subsequent lines, I'd avoid sticking a mutable item like a list in function/method signatures as it could lead to hard-to-debug issues.

Reply:
If you feel strongly I can change it back; doing so requires making the method body more complex to replace the `None` with an empty list to avoid violating the expected type invariant. I was viewing this as a 'if it hurts when you do that, don't do that' kind of thing, given the single line method body. Since this was test code it's also less careful/tolerant of bad input than the actual implementation, but that could be changed.